lib/server.js

```diff
@@ -24,13 +24,27 @@ function handleResponse (opts, req, resp, response) {
             response.headers = {};
         }
 
+        var rh = response.headers;
+
         // Set up CORS
-        response.headers['Access-Control-Allow-Origin'] = '*';
-        response.headers['Access-Control-Allow-Methods'] = 'GET';
-        response.headers['Access-Control-Allow-Headers'] = 'accept, content-type';
+        rh['Access-Control-Allow-Origin'] = '*';
+        rh['Access-Control-Allow-Methods'] = 'GET';
+        rh['Access-Control-Allow-Headers'] = 'accept, content-type';
+
+        if (!/^application\/json/.test(rh['content-type'])) {
+            // Set up security headers
+            // https://www.owasp.org/index.php/List_of_useful_HTTP_headers
+            rh['X-XSS-Protection'] = '1; mode=block';
+            rh['X-Content-Type-Options'] = 'nosniff';
+            rh['X-Frame-Options'] = 'SAMEORIGIN';
+            rh['Content-Security-Policy'] =
+                "default-src 'none'; media-src *; img-src *; style-src *";
```

Comment:
If you're only adding the CSP header to json, you shouldn't have the * modifiers.

Shouldn't you have default-src 'none' here, then and else block that still adds these with a more liberal CSP (I would argue we should enumerate the domains instead of *) and without XFO?